### DISCUSSION TITLE
How to specify in pytest.ini that a .env file should be respected?

### DISCUSSION THREAD

**Discussion by Mithrandir2k18** [pytest-dev/pytest#9261] [Q&A] [answered]
I have a git repo with a src/ subfolder that has all my code and a tests/ subfolder next to it that stores the tests.

I created a `.env` file that specifies this addition to the PYTHONPATH and in `.vscode/settings.json` I set the `python.envFile` variable, and pytest-tests now can find the module imports like `import main` from a file in tests/ for a src/main.py file, without doing it in conftest.py.

However if I run `pytest` from the terminal/in a script, it doesn't respect the .env file and I see no option in the help to specify that either.

How to configure the pytest.ini so that simply running pytest will work, without manually adding the src/ folder to the path in conftest.py?

**Comment by The-Compiler** [comment, marked as answer]
pytest core doesn't come with any support for `.env` files - but there are plugins which add that functionality, such as [pytest-dotenv](https://github.com/quiqua/pytest-dotenv) or [pytest-env](https://github.com/MobileDynasty/pytest-env).

If this is just about `PYTHONPATH`, there's also [pytest-pythonpath](https://github.com/ericpalakovichcarr/pytest-pythonpath) and [pytest-srcpaths](https://github.com/okken/pytest-srcpaths), with native support of a `pythonpath` setting being added to pytest 7.0.0 soon (see #9134).
